### PR TITLE
Updating Clang Version to 11.0.0

### PR DESCRIPTION
**Pull request by codereport**
Follow up PR to: https://github.com/rapidsai/cudf/pull/6695. Performing the same changes but for `rapidsai/cuspatial`

Depends on: https://github.com/rapidsai/integration/pull/304

### CODE REVIEW

File: ci/checks/style.sh [hunk 1]
```diff
@@ -14,6 +14,10 @@ LANG=C.UTF-8
 . /opt/conda/etc/profile.d/conda.sh
 conda activate rapids
 
+gpuci_conda_retry remove --force rapids-build-env rapids-notebook-env
+gpuci_conda_retry install -y "clang=11.0.0"
+gpuci_conda_retry install -y "clang-tools=11.0.0"
```
Comment (dillon-cullinan) [inline ci/checks/style.sh:19]: These lines can be removed. Once the version is updated in integration, our images will come with the correct package.

**Review by dillon-cullinan — CHANGES_REQUESTED** [review]

File: ci/checks/style.sh (hunk 1 above)
Comment (codereport) [reply ci/checks/style.sh]: Done!

**Review by dillon-cullinan — APPROVED** [review]

**Review by thomcom — CHANGES_REQUESTED** [review]
Is it expected that the clang version fails in `gpuCI/cuspatial/check/style` now? Glad to approve when tests pass.

**Comment by codereport** [comment]
> Is it expected that the clang version fails in `gpuCI/cuspatial/check/style` now? Glad to approve when tests pass.

Yes, once https://github.com/rapidsai/integration/pull/304 is merged, I will retrigger CI for all the Clang11 format PRs and they will light up green (and once missing labels have been added)

**Comment by codereport** [comment]
rerun tests

**Comment by codereport** [comment]
rerun tests

**Review by thomcom — APPROVED** [review]

**Comment by thomcom** [comment]
@gpucibot merge

**Pull request merged**